src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -379,8 +379,11 @@ void WbSupervisorUtilities::processImmediateMessages(bool blockRegeneration) {
   mFieldSetRequests.clear();
   if (blockRegeneration)
     return;
+
   WbTemplateManager::instance()->blockRegeneration(false);
-  emit worldModified();
+  const QList<WbRobot *> &robots = WbWorld::instance()->robots();
+  if (robots.contains(mRobot))  // only emit if robot still exists (FieldSetRequest could have regenerated it)
```

Comment:
Can we simplify it this way?
```suggestion
  if (WbWorld::instance()->robots().contains(mRobot))  // only emit if robot still exists (FieldSetRequest could have regenerated it)
```